File: tests/fixtures/erlang/sample.erl
-module(sample).
-export([area/1, area/2]).

-record(point, {x, y}).

-define(PI, 3.14159).
-define(SQUARE(X), (X * X)).

-type shape() :: circle | square.

area(#point{x = X}) ->
    X.

area(circle, Radius) ->
    ?PI * Radius * Radius;
area(square, Side) ->
    ?SQUARE(Side).

File: tests/fixtures/c/sample.c
#include <stdio.h>

#define MAX_ITEMS 32
#define SQUARE(x) ((x) * (x))

struct point {
    int x;
    int y;
};

union value {
    int number;
    char *text;
};

enum color {
    RED,
    GREEN,
};

typedef struct point point_t;

typedef struct {
    int width;
    int height;
} dims_t;

int add(int left, int right);

int add(int left, int right)
{
    return left + right;
}

static char *greeting(void)
{
    return "hello";
}
